### PR TITLE
feat(proto)!: aggregate queries on account address to account_balances and account_summary queries

### PR DESCRIPTION
**Pull request by toteki**
## Description

Similarly to #1188, this PR combines individual queries into aggregate ones to reduce API surface.

Since some of the fields depend heavily on oracle prices and some do not, they were separate into token-only (`account_balances`) and price-sensitive (`account_summary`) queries. The latter will error when oracle is down.

This aggregation also clears the way for future performance optimizations - in particular the `account_summary` fields fetch the same token prices, values, and parameters multiple times during computation. This was unavoidable when they were separate queries, but can be improved soon.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by toteki — COMMENTED** [review]
# Summary of queries affected

Queries with `addr` but no `denom` are simply moved:
- `Supplied(addr)` -> `AccountBalances(addr).Supplied`
- `Collateral(addr)` -> `AccountBalances(addr).Collateral * MarketSummary(denom).UTokenExchangeRate`
- `Borrowed(addr)` -> `AccountBalances(addr).Borrowed`

Note that collateral is a uToken amount, so it needs an exchange rate multiplier to be interpreted properly (we were doing this already, right?)

Queries with both `addr` and `denom` can be extracted from the combined coins:
- `Supplied(addr, denom)` -> `AccountBalances(addr).Supplied.AmountOf(denom)`
- `Collateral(addr, denom)` -> `AccountBalances(addr).Collateral.AmountOf(denom) * MarketSummary(denom).UTokenExchangeRate`
- `Borrowed(addr, denom)` -> `AccountBalances(addr).Borrowed.AmountOf(denom)`

Queries for USD value with `addr` but no `denom` are simply moved:
- `SuppliedValue(addr)` -> `AccountSummary(addr).SuppliedValue`
- `CollateralValue(addr)` -> `AccountSummary(addr).CollateralValue`
- `BorrowedValue(addr)` -> `AccountSummary(addr).BorrowedValue`
- `BorrowLimit(addr)` -> `AccountSummary(addr).BorrowLimit`
- `LiquidationThreshold(addr)` -> `AccountSummary(addr).LiquidationThreshold`

Queries for USD values with both `addr` and `denom` should now be computed with help from `market_summary`:
- `SuppliedValue(addr, denom)` -> `AccountBalances(addr).Supplied.AmountOf(denom) * MarketSummary(denom).OraclePrice`
- `CollateralValue(addr, denom)` -> `AccountBalances(addr).Collateral.AmountOf(denom) * MarketSummary(denom).OraclePrice * MarketSummary(denom).UTokenExchangeRate`
- `BorrowedValue(addr,denom)` -> `AccountBalances(addr).Borrowed.AmountOf(denom) * MarketSummary(denom).OraclePrice`

# Final State of the Proto

The remaining queries (which I have moved into a consistent order in the files affected) are:
- `Params()`
- `RegisteredTokens()`
- `MarketSummary(denom)`
- `AccountBalances(addr)`
- `AccountSummary(addr)`
- `LiquidationTargets()`

File: proto/umee/leverage/v1/query.proto [hunk 1]
```diff
@@ -335,19 +151,84 @@ message QueryMarketSummaryResponse {
...
+  ];
+}
+
+// QueryAccountHealth defines the request structure for the AccountHealth gRPC service handler.
+message QueryAccountHealth {
+  string address = 1;
+}
+
+// QueryAccountHealthResponse defines the response structure for the AccountHealth gRPC service handler.
+message QueryAccountHealthResponse {
```
Comment (robert-zaremba) [inline proto/umee/leverage/v1/query.proto:198]: With `Health` I would expect some kind of metric, maybe 0-100%. Here we are returning aggregated balances. 

Let's change Health in the query, request and response names to something else. Maybe:
1. we can rename `QueryAccountSummary` to `QueryAccountBalances` and use the former name here (renaming `QueryAccountHealth` to `QueryAccountSummary`)?

File: proto/umee/leverage/v1/query.proto [hunk 2]
```diff
@@ -335,19 +151,84 @@ message QueryMarketSummaryResponse {
...
+}
+
+// QueryLiquidationTargets defines the request structure for the
+// LiquidationTargets gRPC service handler.
+message QueryLiquidationTargets {}
+
+// QueryLiquidationTargetsResponse defines the response structure for the
+// LiquidationTargets gRPC service handler.
+message QueryLiquidationTargetsResponse {
+  repeated string targets = 1;
```
Comment (robert-zaremba) [inline proto/umee/leverage/v1/query.proto:233]: please add a comment. What this string contains?

**Review by robert-zaremba — COMMENTED** [review]
Let's think a bit more about the API

File: proto/umee/leverage/v1/query.proto (hunk 2 above)
Comment (toteki) [reply proto/umee/leverage/v1/query.proto:233]: This query is unchanged in this PR - but it's a list of addresses eligible for liquidation

File: proto/umee/leverage/v1/query.proto (hunk 1 above)
Comment (toteki) [reply proto/umee/leverage/v1/query.proto:198]: Good idea - that solves the naming ambiguity

**Review by robert-zaremba — APPROVED** [review]
lgtm

**Pull request merged**